### PR TITLE
Add definitions for scrypt-async v1.2.0

### PR DESCRIPTION
**Pull request by xStrom**
This patch contains the definitions and tests for [scrypt-async](https://github.com/dchest/scrypt-async-js) v1.2.0.

### CODE REVIEW

File: scrypt-async/scrypt-async.d.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Type definitions for scrypt-async v1.2.0
+// Project: https://github.com/dchest/scrypt-async-js
+// Definitions by: Kaur Kuut <https://github.com/xStrom>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module ScryptAsync {
+	interface callbackFunc {
```
Comment (panuhorsmalahti) [inline scrypt-async/scrypt-async.d.ts]: Should be ``CallbackFunc`` or just ``Callback``.

File: scrypt-async/scrypt-async.d.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+// Type definitions for scrypt-async v1.2.0
+// Project: https://github.com/dchest/scrypt-async-js
+// Definitions by: Kaur Kuut <https://github.com/xStrom>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module ScryptAsync {
```
Comment (panuhorsmalahti) [inline scrypt-async/scrypt-async.d.ts]: You should enable importing in Node.js with something like this:
```
declare module "scrypt-async" {
```

**Comment by panuhorsmalahti** [comment]
After the proposed changes are fixed everything looks good for merging.

**Comment by xStrom** [comment]
I implemented the changes.

**Comment by vvakame** [comment]
thanks mate!

**Pull request merged**